src/main/java/org/radarbase/authorizer/webapp/resource/RestSourceUserResource.java

```diff
@@ -35,79 +37,90 @@
+  public ResponseEntity addAuthorizedRestSourceUser(@RequestParam(value = "code") String code,
+      @RequestParam(value = "state") String state) throws URISyntaxException {
+    logger.debug("Add a rest-source user with code {} and state {}", code, state);
+    RestSourceUserPropertiesDTO
+        user = this.restSourceUserService.authorizeAndStoreDevice(code, state);
+    return ResponseEntity
+        .created(new URI("/user/" + user.getId())).body(user);
+  }
 
-        logger.debug("Get all rest source users");
-        return ResponseEntity
-                .ok(this.restSourceUserService.getAllRestSourceUsers());
+  @GetMapping("/users")
+  public ResponseEntity<RestSourceUsers> getAllRestSources(
+      @RequestParam(value = "source-type", required = false) String sourceType) {
+    if (sourceType != null && !sourceType.isEmpty()) {
+      logger.debug("Get all rest source users by type {}", sourceType);
+      return ResponseEntity
+          .ok(this.restSourceUserService.getAllUsersBySourceType(sourceType));
     }
 
-    @GetMapping("/users/{id}")
-    public ResponseEntity<RestSourceUserPropertiesDTO> getRestSourceUserById(
-            @PathVariable String id) {
-        logger.debug("Get rest source user with id {}", id);
-        return ResponseEntity
-                .ok(this.restSourceUserService.getRestSourceUserById(Long.valueOf(id)));
-    }
+    logger.debug("Get all rest source users");
+    return ResponseEntity
+        .ok(this.restSourceUserService.getAllRestSourceUsers());
+  }
 
-    @PostMapping("/users/{id}")
-    public ResponseEntity updateDeviceUser(@Valid @PathVariable String id,
-            @RequestBody RestSourceUserPropertiesDTO restSourceUser) {
-        logger.debug("Requesting to update rest source user");
-        return ResponseEntity
-                .ok(this.restSourceUserService.updateRestSourceUser(Long.valueOf(id), restSourceUser));
-    }
+  @GetMapping("/users/{id}")
+  public ResponseEntity<RestSourceUserPropertiesDTO> getRestSourceUserById(
+      @PathVariable String id) {
+    logger.debug("Get rest source user with id {}", id);
+    return ResponseEntity
+        .ok(this.restSourceUserService.getRestSourceUserById(Long.valueOf(id)));
+  }
 
-    @DeleteMapping("/users/{id}")
-    public ResponseEntity<Void> deleteDeviceUser(@Valid @PathVariable String id) {
-        logger.debug("Requesting to delete rest source user");
-        this.restSourceUserService.revokeTokenAndDeleteUser(Long.valueOf(id));
-        return ResponseEntity
-                .ok().header("user-removed", id).build();
+  @PostMapping("/users/{id}")
+  public ResponseEntity updateDeviceUser(@Valid @PathVariable String id,
+      @RequestBody RestSourceUserPropertiesDTO restSourceUser,
+      @RequestParam(value = "validate", defaultValue = "false") Boolean isValidate) {
+    logger.debug("Requesting to update rest source user");
+    if (validator != null && isValidate && !validator.validate(restSourceUser)) {
```

Comment:
Minor detail. I would switch the `isValidate` first then check if `validator != null`

Reply:
Done